File: PayrollProjectNew/payroll_app/lib/src/features/attendance/model/attendance.dart
import 'package:intl/intl.dart';
import 'package:payroll_app/src/features/employee/model/employee.dart';

class Attendance {
  int? id;
  Employee? employee;
  String? checkIn;
  String? checkOut;

  Attendance({this.id, this.employee, this.checkIn, this.checkOut});

  Attendance.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    employee = json['employee'] != null
        ? Employee.fromJson(json['employee'])
        : null;
    checkIn = json['checkIn'];
    checkOut = json['checkOut'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    if (this.employee != null) {
      data['employee'] = this.employee!.toJson();
    }
    data['checkIn'] = this.checkIn;
    data['checkOut'] = this.checkOut;
    return data;
  }
}

File: Backend/CreateApi/src/main/java/com/shohab/CreateApi/model/Attendance.java
package com.shohab.CreateApi.model;

import jakarta.persistence.*;
import jdk.jfr.Timestamp;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Timestamp
public class Attendance {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @ManyToOne
    @JoinColumn(name = "empid")
    private Employee employee;
    @CreationTimestamp
    private Date checkIn;
    @UpdateTimestamp
    private Date checkOut;


}
